task5/classes/MyFilter.java

```diff
@@ -0,0 +1,31 @@
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebFilter("/*")
+public class MyFilter implements Filter{
+    @Override
+    public void init(FilterConfig filterConfig) {
+    }
+
+    @Override
+    public void doFilter (ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        long start = System.currentTimeMillis();
+        chain.doFilter(request, response);
+        long end = System.currentTimeMillis();
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String method = httpRequest.getMethod();
+        String url = String.valueOf(httpRequest.getRequestURL());
+
+        System.out.println("\n" + method + " - " + url + " - " + (end - start) + "ms");
```

Comment:
Let's learn running cleaner code from the start)
`System.out.printf("Method: %s, url: %s, duration: %dms%n", method, url, end - start);`

This is the way to interpolate strings. `%s` means "a string argument from the ones specified in the following arguments list", `%d` means "numeric value", `%n` is for "new line"
Thus, no string concatenation.
Also, you missed "url" value in your code ;)

Reply:
okay, i got it, but i don't understand what you mean by missing the "url" value... i literally used it in print.